mlflow/store/file_store.py

```diff
@@ -437,37 +450,46 @@ def get_all_tags(self, run_uuid):
             tags.append(self._get_tag_from_file(parent_path, tag_file))
         return tags
 
-    def _list_run_uuids(self, experiment_id, run_view_type):
+    def _lifecycle_stage_valid_for_view_type(self, view_type, lifecycle_stage):
+        if view_type == ViewType.ALL:
+            return True
+        elif view_type == ViewType.ACTIVE_ONLY:
+            return lifecycle_stage == RunInfo.ACTIVE_LIFECYCLE
+        elif view_type == ViewType.DELETED_ONLY:
+            return lifecycle_stage == RunInfo.DELETED_LIFECYCLE
+        else:
+            raise ExecutionException("Invalid view type '%s'" % str(view_type))
+
+    def _list_run_infos(self, experiment_id, view_type):
         self._check_root_dir()
         experiment_dir = self._get_experiment_path(experiment_id, assert_exists=True)
         run_uuids = list_all(experiment_dir, os.path.isdir, full_path=False)
-        if run_view_type == ViewType.ALL:
-            return run_uuids
-        elif run_view_type == ViewType.ACTIVE_ONLY:
-            return [r_id for r_id in run_uuids
-                    if self._get_run_info(r_id).lifecycle_stage == RunInfo.ACTIVE_LIFECYCLE]
-        else:
-            return [r_id for r_id in run_uuids
-                    if self._get_run_info(r_id).lifecycle_stage == RunInfo.DELETED_LIFECYCLE]
+        run_infos = []
+        for r_id in run_uuids:
+            try:
+                run_info = self._get_run_info(r_id)
+                if self._lifecycle_stage_valid_for_view_type(view_type, run_info.lifecycle_stage):
+                    run_infos.append(run_info)
+            except ResourceNotFoundException as rnfe:
+                # trap malformed exception and log warning
+                logging.warning("Malformed run '%s'. Detailed error %s", r_id, str(rnfe))
```

Comment:
Should we be using the `exc_info` kwarg for `logging.warning`?

Reply:
lemme try doing that. would be awesome to `exc_info` real exception. When i tried it first, the library did not like any unexpected arguments passed in `kwargs`